crates/ruff_linter/src/rules/flake8_pyi/rules/unused_private_type_definition.rs

```diff
@@ -185,13 +192,22 @@ pub(crate) fn unused_private_type_var(
         let Expr::Call(ast::ExprCall { func, .. }) = value.as_ref() else {
             continue;
         };
-        if !checker.semantic().match_typing_expr(func, "TypeVar") {
-            continue;
-        }
+
+        let semantic = checker.semantic();
+
+        let typevarlike_kind = match func {
+            f if semantic.match_typing_expr(f, "TypeVar") => "TypeVar",
+            f if semantic.match_typing_expr(f, "ParamSpec") => "ParamSpec",
+            f if semantic.match_typing_expr(f, "TypeVarTuple") => "TypeVarTuple",
+            _ => {
+                continue;
+            }
+        };
```

Comment:
I might write this like:
```rust
let typevarlike_kind = if semantic.match_typing_expr(f, "TypeVar")  {
   "TypeVar"
} else if ...
```

But I don't know that what I'm proposing is actually _better_, I just hadn't seen this pattern before :)